pkg/controller/controller_ref_manager.go

```diff
@@ -213,6 +213,58 @@ func (m *VirtualMachineControllerRefManager) ClaimVirtualMachineInstances(vmis [
 	return claimed, utilerrors.NewAggregate(errlist)
 }
 
+// ClaimVirtualMachines tries to take ownership of a list of VirtualMachines.
```

Comment:
It's great that we can leverage the `LabelSelector` pattern to do attach and detach.  However, what is the use-case for attach?  I don't recall discussing it in the design doc so I want to make sure I understand how this would look.

Reply:
> However, what is the use-case for attach? I don't recall discussing it in the design doc so I want to make sure I understand how this would look.

Ha, good point. I don't actually have a reasonable use case for why someone would want to attach a VM after detaching it.

If someone merely wanted to investigate something on a VM then return it to the pool, they could pause the pool instead. That functionality exists now.

Unless there's a good reason to support attach, I'm actually in favor of removing it for now. Lets discuss it in sig scale community meeting.

Reply:
I also don't see a user-facing `attach` use-case. However I would expect for consistency that a controller would recognize suddenly matching labels, accept the VM as part of the pool and take scaling actions according to the definition.

Reply:
> I also don't see a user-facing attach use-case. However I would expect for consistency that a controller would recognize suddenly matching labels, accept the VM as part of the pool and take scaling actions according to the definition.

that could have unexpected destructive results for VMs.

For example, a VM magically matches a random Pool's selector, the VM get's adopted, then the Pool updates that VM in a way that destroys the VM's previous state. That would be really confusing.

For now, I've removed the adoption logic from the pool entirely. The pool will only detach VMs that no longer have the match labels and reattachment is not possible after detachment.